core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.persistence.cassandra.journal
+
+import akka.persistence.cassandra.query.EventsByPersistenceIdStage.Extractors
+import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
+import akka.persistence.cassandra.session.scaladsl.CassandraSession
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import com.datastax.driver.core.policies.{ LoggingRetryPolicy, RetryPolicy }
+import com.datastax.driver.core._
+import java.lang.{ Long => JLong }
+
+import scala.language.implicitConversions
+import akka.persistence.cassandra._
+import akka.Done
+import akka.event.LoggingAdapter
+import akka.persistence.cassandra.journal.CassandraDeletion.{ MessageId, PartitionInfo }
+
+import scala.collection.immutable
+import scala.concurrent.{ ExecutionContext, Future }
+
+object CassandraDeletion {
+  private case class PartitionInfo(partitionNr: Long, minSequenceNr: Long, maxSequenceNr: Long)
+  private case class MessageId(persistenceId: String, sequenceNr: Long)
+}
+
+trait CassandraDeletion extends CassandraStatements {
```

Comment:
this is exposing too much things as public if it's intended to be public api

Reply:
well, perhaps it's just the prepared statements
however, might be worth defining a trait (interface) without implementation as the public api